### PR TITLE
Install tier-0 reliability: bounded privileged execution, VHID ordering guard, 0-byte config fix

### LINKED ISSUE #927: Installer: privileged osascript fallback can hang forever with no timeout or error surface
## Symptom (fresh-install session, 2026-07-02, v1.0.0 RC 812245f0)

The osascript privileged fallback ("KeyPath needs to repair the VirtualHID services.") ran a script that blocked on `launchctl kickstart -k system/com.keypath.karabiner-vhiddaemon` for a daemon whose binary was never installed. The root-owned script hung for **18+ minutes** (until manually killed) with no timeout, no error dialog, and no log entry after launch.

- 05:18:01.712 PRIVILEGED-TRIGGER → user enters admin password at 05:18:09
- osascript PID 98542 / bash 98628 / launchctl 98650 still alive at 05:36
- User-visible result: typed an admin password and got literally nothing — no success, no failure, system silently stays broken.

## Fix direction

- Hard timeout on every privileged shell fallback (both the osascript wrapper and the inner script steps — `launchctl kickstart` especially).
- On timeout: kill the process tree, surface an actionable error in the wizard, log the failing step.
- Precondition checks before kickstart (binary exists) — see companion ordering issue.

## Acceptance
No privileged fallback can outlive a bounded timeout; failures surface in UI + log.

Part of the 1.0 install-QA tier-0 batch. Related: #749, #747.

### LINKED ISSUE #928: Installer: repair registers VHID LaunchDaemons before Karabiner driver payload exists → crash-loop the wizard can't fix
## Symptom (fresh-install session, 2026-07-02, v1.0.0 RC 812245f0)

While the user was parked on the Input Monitoring wizard page, the auto-repair path (`installRequiredRuntimeServices` → `repairVHIDDaemonServices` fallback) wrote and bootstrapped the VirtualHID LaunchDaemon plists **before the Karabiner driver payload was ever installed** (the "Karabiner Driver Setup" wizard page was never reached).

End state verified on disk:
- `/Library/Application Support/org.pqrs` missing; hidden Manager app missing; `systemextensionsctl list` → 0 extensions
- VHID daemons registered but crash-looping: `state=spawn, pid=nil, lastExit=78` (EX_CONFIG — binaries don't exist)
- Perpetual validation loop every ~4s (122+ cycles: `missingComponents: karabinerDriver, karabinerDaemon, vhidDeviceRunning`)

This produced a state the wizard UI could not recover from.

## Fix direction

- Enforce install ordering in the repair path, same as the wizard's own sequencing: never register/bootstrap VHID daemons unless the driver payload + binaries exist on disk.
- Repair should validate preconditions and route the user to the driver-install step instead of registering broken services behind their back.

## Acceptance
Fresh install with no driver payload never ends up with registered crash-looping VHID daemons; repair either installs prerequisites first or surfaces "driver required" guidance.

Part of the 1.0 install-QA tier-0 batch. Related: #749, #855.

### LINKED ISSUE #929: Config: default keypath.kbd created as 0-byte file; hot-reload validates the app's own write and error-beeps during setup
## Symptom (fresh-install session, 2026-07-02, v1.0.0 RC 812245f0)

- Session 1: `createDefaultIfMissing` logged "Ensuring default user config" (05:14:41.358) but the file was never created — 3× ERROR `keypath.kbd couldn't be opened... no such file`.
- Session 2 (after relaunch): the file was created as a **0-byte file** (05:17:01), rewritten still-empty at 05:17:31, and remained 0 bytes for the rest of the session.
- The hot-reload watcher treated the app's own write as an external change, validated the empty file (`Exactly one defsrc must exist; found none`), and played an **audible error beep** mid-setup (05:17:01.291; tink-then-beep at 05:17:32).

Impact: mysterious error sounds during first-run setup, and kanata could never start even if services were healthy.

## Fix direction

- Make default-config creation atomic (write temp + rename) and verify non-empty content after write; log + retry on failure.
- Suppress hot-reload validation/sounds for writes originating from the app itself (or during wizard/setup).
- Add a regression test: fresh HOME → createDefaultIfMissing → file exists, non-empty, passes validation.

Part of the 1.0 install-QA tier-0 batch. Related: #736.

### LINKED ISSUE #930: Helper: freshly-approved helper times out twice over XPC (2×30s) → cascades to osascript admin prompt mid-permission-flow
## Symptom (fresh-install session, 2026-07-02, v1.0.0 RC 812245f0)

Immediately after the privileged helper was approved and verified functional (05:15:17 "Helper installed and verified functional"), the very next privileged operations both timed out:

- 05:17:01 → 05:17:31: `installRequiredRuntimeServices` XPC timeout after 30s
- 05:17:31 → 05:18:01: fallback `repairVHIDDaemonServices` XPC timeout after another 30s (HelperManager+RequestHandlers.swift:92)
- Result: **1 minute of dead air**, then the osascript fallback fired **admin password prompt #2** ("KeyPath needs to repair the VirtualHID services") — landing while the user was mid-grant in System Settings for Input Monitoring (SecurityAgent frontmost 05:18:01–05:18:09).

The helper's whole purpose is avoiding osascript prompts; it fell back anyway, at the worst possible moment.

## Investigate / fix direction

- Why did the just-verified helper not answer? (busy? blocked on its own launchctl call? connection invalidated after Login Items approval?)
- Progress/feedback in the wizard while a privileged call is in flight (no silent 30s waits).
- Don't fire an interactive admin prompt while the user is mid-task in System Settings; queue or surface non-modally.
- Consider retry-once-with-fresh-connection before falling back to osascript.

Part of the 1.0 install-QA tier-0 batch. Related: #748, #749, #927, #928.

### PR DESCRIPTION
**Pull request by malpern**
## Summary

Fixes the tier-0 reliability failures from the 2026-07-02 fresh-install QA session (the install dead-ended with a root repair script hung 18+ minutes, crash-looping VHID daemons, and an unparseable 0-byte config).

### #927 — privileged execution can never hang unbounded
- **Helper:** `HelperService.run()` now delegates to a shared `BoundedProcess` runner (KeyPathCore): hard timeout (60s default, 15s for `launchctl kickstart -k`, 300s for download/install), terminate→SIGKILL on expiry, concurrent pipe draining (also fixes a latent full-pipe deadlock for chatty commands like `launchctl print`).
- **App:** `PrivilegedCommandRunner` osascript/sudo paths are bounded (900s / 360s). `PrivilegedExecutor` now delegates instead of being a third unbounded copy.
- **Root script self-watchdog:** killing the osascript client can't reach the root child under the security trampoline, so every privileged temp script bounds ITSELF (300s), with identity re-verified via the unique script name before the self-kill.
- **`kp_timeout` batch watchdog:** batch scripts get a shell-level per-command watchdog; the known-hazard `kickstart -k` steps are wrapped at 15s. Parentage is re-checked before the SIGKILL so a root script can never kill a recycled PID.

### #928 — never register VHID services without the driver payload
Both the helper (`installOrRepairVHIDServices`) and the sudo path (`ServiceBootstrapper.repairVHIDDaemonServices`, via the canonical `VHIDDeviceManager.detectInstallation()/detectActivation()` predicates) now refuse to bootstrap/kickstart VHID daemons when the Karabiner payload is missing — previously this produced exit-78 spawn-loops and the kickstart hang, in a state the wizard couldn't repair. The sudo-path guard also fails *before* showing an admin password prompt for a repair that cannot succeed, and the router logs the helper's actionable error instead of swallowing it.

### #929 — 0-byte keypath.kbd eliminated at all three sources
- Helper VHID scaffolding no longer `touch`es the config (kept: directory scaffold + ownership repair of an existing root-owned config).
- **kanata-launcher** no longer creates an empty config as root on daemon start (the twin the log analysis missed initially).
- App-side self-heal: `createInitialConfigIfNeeded` and `reload()` treat a 0-byte/whitespace-only config as missing and rewrite the default — **conservatively**: an unreadable or non-UTF-8 file is never overwritten (data-loss guard).

### #930 — helper XPC timeouts
Root cause was the same kickstart hang inside the helper (it shares `installOrRepairVHIDServices`), so #927/#928 fix it at the source: the helper now answers fast with a real error instead of hanging into the app's 30s XPC timeout twice and cascading to a surprise osascript password prompt mid-permission-flow.

## Review
8-angle review (line-scan, removed-behavior, cross-file, reuse/simplification/efficiency, altitude, conventions) ran pre-PR; all confirmed findings addressed in the second commit (notably: launcher 0-byte twin, ownership-repair regression, dialog-time-in-timeout, unreadable-as-empty data-loss risk, root-script orphaning, duplicate bounded-run implementations).

## Tests
- New `PrivilegedCommandRunnerBatchTests`: prelude generation + real kp_timeout kill behavior (0.3s fractional deadline).
- New `ConfigurationServiceTests` cases: empty config rewritten; non-empty config preserved.
- Note: 3 pre-existing `ConfigurationServiceTests` cases fail on machines with real user rule stores (fail identically on master; clean on CI) — test-isolation fix tracked separately.

Closes #927. Closes #928. Closes #929. Closes #930.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Sources/KeyPathAppKit/Core/PrivilegedOperationsRouter.swift [hunk 1]
```diff
@@ -112,6 +112,12 @@ public final class PrivilegedOperationsRouter {
                     try await sudoRepairVHIDServices()
                 }
             } catch {
+                // Keep the helper's actionable reason (e.g. "driver payload is
+                // not installed") visible — the sudo fallback's generic failure
+                // would otherwise bury it (#928).
+                AppLogger.shared.warn(
+                    "⚠️ [PrivilegedOperationsRouter] Helper install failed (\(error.localizedDescription)) — falling back to sudo path"
+                )
                 try await sudoInstallRequiredRuntimeServices()
```
[the remark this answers is not in the document]
Comment (malpern) [reply Sources/KeyPathAppKit/Core/PrivilegedOperationsRouter.swift]: Fixed in b30a89a1 — the catch now rethrows with both the helper's actionable reason and the fallback failure when the sudo path also fails.
